src/js/media-container.js

```diff
@@ -145,6 +145,94 @@ template.innerHTML = `
 
 const MEDIA_UI_ATTRIBUTE_NAMES = Object.values(MediaUIAttributes);
 
+// Watch for child adds/removes and update the media element if necessary
+const mutationCallback = (mutationsList) => {
+  for (let mutation of mutationsList) {
+    /** @type MediaContainer */
+    const container = mutation.target;
+    const media = container.media;
+
+    if (mutation.type === 'childList') {
+      // Media element being removed
+      mutation.removedNodes.forEach((node) => {
+        // Is this a direct child media element of media-controller?
+        // TODO: This accuracy doesn't matter after moving away from media attrs.
+        // Could refactor so we can always just call 'dispose' on any removed media el.
+        if (node.slot == 'media') {
+          // Check if this was the current media by if it was the first
+          // el with slot=media in the child list. There could be multiple.
+          let previousSibling =
+            mutation.previousSibling &&
+            mutation.previousSibling.previousElementSibling;
+
+          // Must have been first if no prev sibling or new media
+          if (!previousSibling || !media) {
+            container.mediaUnsetCallback(node);
+          } else {
+            // Check if any prev siblings had a slot=media
+            // Should remain true otherwise
+            let wasFirst = previousSibling.slot !== 'media';
+            while (
+              (previousSibling = previousSibling.previousSibling) !== null
+            ) {
+              if (previousSibling.slot == 'media') wasFirst = false;
+            }
+            if (wasFirst) container.mediaUnsetCallback(node);
+          }
+        }
+      });
+
+      // Controls or media element being added
+      // No need to inject anything if media=null
+      if (media) {
+        mutation.addedNodes.forEach((node) => {
+          if (node == media) {
+            // Update all controls with new media if this is the new media
+            container.handleMediaUpdated(media).then((media) =>
+              container.mediaSetCallback(media)
+            );
+          }
+        });
+      }
+    }
+  }
+};
+
+const defaultBreakpoints = 's:0 sm:384 m:576 ml:960 l:1280 xl:1600 fhd:1920 qhd:2560 uhd:3840';
```

Comment:
Though: it is a bit weird that these represent the minimum for the size. Though, I can see how that simplifies some things, like having the uhd match the 4k-uhd dimensions as the minimum and that it doesn't need to specify Infinity or something.
I think it's a bit easier to read these and reason about them if they are maximums, but then it leaves how to represent the last item, though, I guess could use Infinity directly.
```
s:384 sm:576 m:960 ml:1280 l:1600 xl:1920 fhd:2560 qhd:3840 uhd:Infinity
```
That said, I'm not sure the difference is big enough compared to the downsides.

Reply:
one other solution which would also solve the multiple CSS selector issue is to be able to define ranges for each label but it's a little more verbose.

```
s:0:384 sm:384:576 smaller:0:479 bigger:480:1920
```

Reply:
It's probably fine as is, I'm not sure if the alternatives are any better. Just pointing out, that to me, it's a bit easier to know that the config is the maximum as it feels a bit different from setting the CSS directly or usage via bootstrap/tailwind.